app/src/server/trpc/routers/service.ts

```diff
@@ -36,6 +37,25 @@ export const serviceRouter = router({
       return artifactList;
     }),
 
+  getDataRequirements: publicProcedure
+    .input(z.object({ resourceType: z.enum(['Measure', 'Library']), id: z.string() }))
+    .query(async ({ input }) => {
+      const res = await fetch(
+        `${process.env.NEXT_PUBLIC_MRS_SERVER}/${input.resourceType}/${input.id}/$data-requirements`
+      );
+      const resource = await res.json();
+      if (resource?.resourceType === 'OperationOutcome') {
+        const eMessage = resource?.issue[0]?.details?.text;
+        throw new TRPCError({
+          code: 'INTERNAL_SERVER_ERROR',
+          message: eMessage
+        });
+      }
+      return {
+        Library: resource as fhir4.Library
+      } as const;
```

Comment:
In other procedures, it was helpful to return an object since there were multiple things, but here since we are just returning a Library I think you can just do `return resource as fhir4.Library`. You will need to change the way you access `resourceType` in the other file.